wd_helper.au3

```diff
@@ -2376,6 +2382,77 @@ Func _WD_GetElementByRegEx($sSession, $sMode, $sRegExPattern, $sRegExFlags = "",
 	Return SetError(__WD_Error($sFuncName, $iErr), $iRow, $vResult)
 EndFunc   ;==>_WD_GetElementByRegEx
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_Storage
+; Description ...:
+; Syntax ........: _WD_Storage($sSession,  $sKey[,  $vValue = Default[,  $nType = Default]])
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $vKey                - Key to manipulate.
+;                  $vValue              - [optional] Value to store.
+;                  $nType               - [optional] Storage type. Default is $_WD_STORAGE_Local
+; Return values .: Success - Response from _WD_ExecuteScript() and sets @error to $_WD_ERROR_Success
+;                  Failure - @error set to $_WD_ERROR_InvalidArgue if there are no matches OR
+;                            Response from _WD_ExecuteScript() and sets @error to value returned from _WD_ExecuteScript()
+; Author ........: Danp2
+; Modified ......:
+; Remarks .......:	See below for special conditions --
+;
+;					| Parameter | Condition | Action                     |
+;					|-----------|-----------|----------------------------|
+;					| $vKey     | Numeric   | Return name of the Nth key |
+;					| $vKey     | Null      | Clear storage              |
+;					| $vValue   | Null      | Remove key from storage    |
+;
+; Related .......:
+; Link ..........: https://developer.mozilla.org/en-US/docs/Web/API/Storage
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_Storage($sSession, $vKey, $vValue = Default, $nType = Default)
+	Local Const $sFuncName = "_WD_Storage"
+	Local $sParams, $vResult = '', $iErr = $_WD_ERROR_Success
+	Local $bIsKeyNull = (IsKeyword($vKey) = $KEYWORD_NULL), $bIsValueNull = (IsKeyword($vValue) = $KEYWORD_NULL)
+	Local Const $sParameters = 'Parameters:   Key=' & ($bIsKeyNull ? "Null" : $vKey) & '   Value=' & ($bIsValueNull ? "Null" : $vValue) & '   Type=' & $nType
+	
+	If $nType = Default Or $nType < $_WD_STORAGE_Local Or $nType > $_WD_STORAGE_Session Then $nType = $_WD_STORAGE_Local
+
+	Local $sTarget = ($nType = $_WD_STORAGE_Local) ? "window.localStorage" : "window.sessionStorage"
+	Local $sJavaScript = 'return ' & $sTarget 
+
+	Select
+	Case $bIsKeyNull ; Empty storage
+		If $vValue = Default Then
+			$sJavaScript &= '.clear()' 
+			$sParams = $_WD_EmptyDict
+		Else
+			$iErr = $_WD_ERROR_InvalidArgue			
+		EndIf
+
+	Case $vValue = Default ; Retrieve key
+		If IsNumber($vKey) Then
+			$sJavaScript &= '.key(arguments[0])'
+			$sParams = String($vKey)
```

Comment:
why not like this:
```autoit
$sParams = '"' & String($vKey) & '"'
```

Reply:
> something is wrong with code formating in yours example 

Thanks... just a copy/paste issue. Fixed.

Reply:
> Why not like this:
> 
> ```autoit
> $sParams = '"' & String($vKey) & '"'
> ```

From [here](https://developer.mozilla.org/en-US/docs/Web/API/Storage/key) --

![image](https://user-images.githubusercontent.com/7101313/178163120-dd074d5f-94b7-4122-8146-332948b97955.png)

Reply:
so now I wonder if here:
```autoit
	Case $vKey And $vValue ; Set key
		$sJavaScript &=  '.setItem(arguments[0], arguments[1])'
		$sParams = '"' & $vKey & '","' & $vValue & '"'
```

does this `$vValue` can be set as number ?

Reply:
but looking here:
https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem

I see that in this case `$vValue` must to be passed as string.

You can click as resolved.